pkg/util/azureclient/azuresdk/armauthorization/roledefinitions.go

```diff
@@ -0,0 +1,26 @@
+package armauthorization
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/authorization/armauthorization/v3"
+)
+
+type RoleDefinitionsClient interface {
+	GetByID(ctx context.Context, roleID string, options *armauthorization.RoleDefinitionsClientGetByIDOptions) (armauthorization.RoleDefinitionsClientGetByIDResponse, error)
+}
+
+type roleDefinitionsClient struct {
+	armauthorization.RoleDefinitionsClient
+}
+
+var _ RoleDefinitionsClient = &roleDefinitionsClient{}
+
+func NewRoleDefinitionsClient(credential azcore.TokenCredential, options *arm.ClientOptions) (RoleDefinitionsClient, error) {
```

Comment:
In general, we should return structs, not interfaces. It makes code more flexible and robust. https://duncanleung.com/go-idiom-accept-interfaces-return-types/

Reply:
Throughout the codebase this is the norm to use the Interface to hide the Client function implementation that is not needed. I am happy to use the concrete struct type, but can you please point how to approach this here?